components/collector/tests/source_collectors/source_collector_test_case.py

```diff
@@ -75,10 +78,14 @@ def __get_response(json_return_value, json_side_effect, content, text, headers,
         return get_response
 
     @staticmethod
-    def __post_response(json_return_value) -> AsyncMock:
+    def __post_response(json_return_value, json_side_effect: list = None) -> AsyncMock:
         """Create the mock post response."""
         post_response = AsyncMock()
-        post_response.json.return_value = json_return_value
+        if json_side_effect:
+            if not json_return_value:  # put the last side effect into return value, if it was not already given
+                json_return_value = json_side_effect[-1]
```

Comment:
The comment tells me what the code does, but I can read that for myself. It however doesn't explain *why* it does that.